components/server/src/workspace/env-var-resolver.ts

```diff
@@ -25,7 +25,7 @@ export interface ResolvedEnvVars {
 }
 
 @injectable()
-export class EnvVarService {
+export class EnvVarResolver {
```

Comment:
Why don't we merge it in EnvVarService? We also discussed that it should not user services but DBs directly instead. cc @svenefftinge

Reply:
Yes, we should keep the resolve method in `EnvVarService` sorry if that was not clear enough from the description in the issue.

Reply:
Logic of the `resolve` method is related to workspace, method parameter is a `workspace` object and it also takes into account project env vars, so I think it should be part of workspaceService, right now envVarService is exclusive for only user env vars and is inside the user folder

> We also discussed that it should not use services but DBs directly instead.

This will require a duplication of code :thinking:

Reply:
The idea is that the envVarService is responsible for all things env vars and would store them on both users and projects using the DB interface as mentioned in the linear issue: `The env var service encapsulates write and read access of env vars on user and project.`

> Logic of the resolve method is related to workspace, method parameter is a workspace object

Good point! If you want to avoid passing in a workspace object, you could limit the argument for resolve to what is actually needed for a resolution. I.e. instead of passing a workspace object we could pass `requestorId, userId, projected, cloneURL, workspaceKind`.